src/libraries/Microsoft.NET.WebAssembly.Webcil/src/Webcil/WebcilWasmWrapper.cs

```diff
@@ -0,0 +1,166 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Collections.Immutable;
+using System.Reflection.PortableExecutable;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.NET.WebAssembly.Webcil;
+
+//
+// This is a pretty dumb wrapper.  It assumes that the entire wasm module is going to be unchanging,
+// except for the data section which has 2 passive segments.  segment 0 is 4 bytes and contains the
+// length of the webcil payload.  segment 1 is of a variable size and contains the webcil payload.
+//
+// the unchanging parts are stored as a "prefix" and "suffix" which contain the bytes for the following
+// WAT program, split into the parts that come before the data section, and the bytes that come after:
+//
+// (module
+//  (data "\0f\00\00\00") ;; data segment 0: payload size as a 4 byte LE uint32
```

Comment:
I still don't understand why the payload size is in a data segment instead of in a global or the return value of a callable export, honestly

Reply:
because I don't want to edit a global section or a func section of a module

Reply:
OK, so it's explicitly a zone that is safe to patch and is of a known size

Reply:
The data section is the easiest one to parse, so if we're going to do this "let the runtime read the bytes and pull out what it needs" thing, then it is ideal to keep everything variable in the data section.

To be honest the runtime doesn't really need the webcil payload size, so `webcilGetSize` could use a global.  But then I'd probably need to pull in a real WebAssembly parser/writer like wa-info, and since it hasn't yet landed in the runtime repo, I didn't want to be the one to do it 😁